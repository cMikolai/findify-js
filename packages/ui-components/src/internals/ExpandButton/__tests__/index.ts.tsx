import * as React from 'react';
import { ExpandButton, Props } from '..';

describe('<ExpandedButton />', () => {
  it('renders correctly', () => {
    const properties: Props[] = [
      { label: 'foo', expanded: true },
      { label: 'bar', expanded: false },
    ];
    properties.forEach((props: Props) => {
      const component = render(<ExpandButton {...props} />);
      expect(component).toMatchSnapshot();
    });
  });

  it('attaches and calls onClick event handler', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <ExpandButton expanded label="foo" onClick={onClick} />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.simulate('click');
    expect(onClick.mock.calls.length).toBe(1);
  });
});
