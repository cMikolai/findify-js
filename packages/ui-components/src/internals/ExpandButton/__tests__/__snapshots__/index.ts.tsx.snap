// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandedButton /> attaches and calls onClick event handler 1`] = `
<button
  className="wrap"
  onClick={[MockFunction]}
>
  <span
    className="sign"
  >
    -
  </span>
  foo
</button>
`;

exports[`<ExpandedButton /> renders correctly 1`] = `
<button
  class="wrap"
>
  <span
    class="sign"
  >
    -
  </span>
  foo
</button>
`;

exports[`<ExpandedButton /> renders correctly 2`] = `
<button
  class="wrap"
>
  <span
    class="sign"
  >
    +
  </span>
  bar
</button>
`;
